community-factory/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{env, near_bindgen, require, AccountId, Gas, NearToken, Promise};
+
+const CODE: &[u8] = include_bytes!("../../res/devhub_community.wasm");
+const INITIAL_BALANCE: NearToken = NearToken::from_near(2);
+const PUBKEY_STR: &str = "ed25519:4deBAvg1S4MF7qe9GBDJwDCGLyyXtJa73JnMXwyG9vsB";
+const DEVHUB: &str = "devgovgigs.near";
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, Default)]
+pub struct Contract {}
+
+#[near_bindgen]
+impl Contract {
+    #[payable]
+    pub fn create_community_account(&mut self, community: String) -> Promise {
+        let devhub_account: AccountId = DEVHUB.parse().unwrap();
+        require!(
+            env::predecessor_account_id() == devhub_account,
+            "Can only be called from DevHub contract"
+        );
+        require!(
+            env::attached_deposit() == INITIAL_BALANCE,
+            "Require 2 NEAR to create community account"
+        );
+
+        let community_account_id: AccountId =
+            format!("{}.{}", community, env::current_account_id()).parse().unwrap();
```

Comment:
This is also a good candidate for a dedicated method in near-account-id: https://github.com/near/near-account-id-rs/issues/23

Reply:
Yeah this helper is also neat